python/pyspark/broadcast.py

```diff
@@ -75,7 +75,13 @@ def __init__(self, sc=None, value=None, pickle_registry=None, path=None):
             self._path = path
 
     def dump(self, value, f):
-        pickle.dump(value, f, 2)
+        try:
+            pickle.dump(value, f, 2)
+        except pickle.PickleError:
+            raise
+        except Exception as e:
+            msg = "Could not serialize broadcast: " + e.__class__.__name__ + ": " + e.message
+            raise pickle.PicklingError(msg)
```

Comment:
it is indeed a good practice to log when wrapping exception is more contextual one. logging.exception(msg) will be perfect